src/components/calcite-accordion/calcite-accordion.tsx

```diff
@@ -0,0 +1,212 @@
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify the theme of dropdrown, defaults to light */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of dropdrown, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** specify the appearance, defaults to default */
+  @Prop({ mutable: true, reflect: true }) appearance: "default" | "minimal" =
+    "default";
+
+  /** specify the placement of the icon, defaults to end */
+  @Prop({ mutable: true, reflect: true }) iconPosition: "start" | "end" = "end";
+
+  /** specify the selection mode - defaults to multi */
+  @Prop({ mutable: true, reflect: true }) selectionMode:
+    | "multi"
+    | "single"
+    | "single-persist" = "multi";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteAccordionItemHasChanged: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // validate props
+    let appearance = ["default", "minimal"];
+    if (!appearance.includes(this.appearance)) this.appearance = "default";
+
+    let iconPosition = ["start", "end"];
+    if (!iconPosition.includes(this.iconPosition)) this.iconPosition = "end";
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+
+    let scale = ["s", "m", "l"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+
+    let selectionMode = ["multi", "single", "single-persist"];
+    if (!selectionMode.includes(this.selectionMode))
+      this.selectionMode = "multi";
+  }
+
+  componentDidLoad() {
+    if (!this.sorted) {
+      this.items = this.sortItems(this.items);
+      this.sorted = true;
+    }
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host dir={dir} id={this.accordionId} tabindex="-1">
+        <div class="calcite-accordion-wrapper" role="menu">
+          <slot />
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteAccordionItemKeyEvent") calciteAccordionItemKeyEvent(
+    item: CustomEvent
+  ) {
+    let e = item.detail.item;
```

Comment:
one letter variables hurt readability. Please give little 'e' a real name.
http://devhumor.com/media/naming-is-the-most-difficult-part
😄